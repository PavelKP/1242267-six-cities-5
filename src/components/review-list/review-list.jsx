import React from 'react';
import Review from '../review/review';
import PropTypes from 'prop-types';
import {reviewsPropTypes} from '../../prop-types/prop-types';
import withCommentsLoading from '../../hocs/with-comments-loading/with-comments-loading';

const ReviewList = (props) => {
  const {reviews, loading} = props;

  return (
    loading
      ? <h3>Loading...please wait</h3>
      : <React.Fragment>
        <h2 className="reviews__title">Reviews &middot;
          <span className="reviews__amount">{reviews.length}</span>
        </h2>
        <ul className="reviews__list">
          {reviews.map((review) => (
            <Review review={review} key={`review-${review.id}`} />
          ))}
        </ul>
      </React.Fragment>
  );
};

ReviewList.propTypes = {
  reviews: reviewsPropTypes.reviews,
  loading: PropTypes.bool.isRequired,
};

export {ReviewList};
export default withCommentsLoading(ReviewList);
