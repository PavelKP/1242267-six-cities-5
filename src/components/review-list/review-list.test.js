import React from 'react';
import renderer from 'react-test-renderer';
import {ReviewList} from './review-list';
import configureStore from 'redux-mock-store';
import {Provider} from 'react-redux';
import mockStoreData, {reviews} from '../../__mocks__/mock-store';
import {Router} from 'react-router-dom';
import browserHistory from '../../browser-history';

const mockStore = configureStore([]);

describe(`<ReviewList /> render`, () => {

  let store;

  beforeEach(() => {
    store = mockStore(mockStoreData);
  });

  it(`Should ReviewList with main cards render correctly`, ()=> {
    const tree = renderer.create(
        <Provider store={store}>
          <Router history={browserHistory}>
            <ReviewList
              reviews={reviews}
              loading={false}
            />
          </Router>
        </Provider>)
        .toJSON();

    expect(tree).toMatchSnapshot();
  });

});
