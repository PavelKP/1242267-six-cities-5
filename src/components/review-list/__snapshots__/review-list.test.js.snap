// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ReviewList /> render Should ReviewList with main cards render correctly 1`] = `
Array [
  <h2
    className="reviews__title"
  >
    Reviews Â·
    <span
      className="reviews__amount"
    >
      2
    </span>
  </h2>,
  <ul
    className="reviews__list"
  >
    <li
      className="reviews__item"
    >
      <div
        className="reviews__user user"
      >
        <div
          className="reviews__avatar-wrapper user__avatar-wrapper"
        >
          <img
            alt="Reviews avatar"
            className="reviews__avatar user__avatar"
            height="54"
            src="https://assets.htmlacademy.ru/intensives/javascript-3/avatar/6.jpg"
            width="54"
          />
        </div>
        <span
          className="reviews__user-name"
        />
      </div>
      <div
        className="reviews__info"
      >
        <div
          className="reviews__rating rating"
        >
          <div
            className="reviews__stars rating__stars"
          >
            <span
              style={
                Object {
                  "width": "100%",
                }
              }
            />
            <span
              className="visually-hidden"
            >
              Rating
            </span>
          </div>
        </div>
        <p
          className="reviews__text"
        >
          The best hotel ever!
        </p>
        <time
          className="reviews__time"
          dateTime="2020-11-20"
        >
          November 2020
        </time>
      </div>
    </li>
    <li
      className="reviews__item"
    >
      <div
        className="reviews__user user"
      >
        <div
          className="reviews__avatar-wrapper user__avatar-wrapper"
        >
          <img
            alt="Reviews avatar"
            className="reviews__avatar user__avatar"
            height="54"
            src="https://assets.htmlacademy.ru/intensives/javascript-3/avatar/3.jpg"
            width="54"
          />
        </div>
        <span
          className="reviews__user-name"
        />
      </div>
      <div
        className="reviews__info"
      >
        <div
          className="reviews__rating rating"
        >
          <div
            className="reviews__stars rating__stars"
          >
            <span
              style={
                Object {
                  "width": "80%",
                }
              }
            />
            <span
              className="visually-hidden"
            >
              Rating
            </span>
          </div>
        </div>
        <p
          className="reviews__text"
        >
          Home is amazing. It's like staying in a museum. The rooms, furnishings and artworks are incredible. The views of My Vesuvius,
        </p>
        <time
          className="reviews__time"
          dateTime="2020-11-04"
        >
          November 2020
        </time>
      </div>
    </li>
  </ul>,
]
`;
